#! /bin/sh /usr/share/dpatch/dpatch-run
## 20_http_double_free.dpatch by Paul Brossier <piem@debian.org>
##
## DP: Initialise host and path buffers to NULL, avoiding double free when http
## DP: port parsing fails 

@DPATCH@
diff -urNad alsaplayer-0.99.80~/alsaplayer/CorePlayer.h alsaplayer-0.99.80/alsaplayer/CorePlayer.h
--- alsaplayer-0.99.80~/reader/http/http.c	2009-09-16 01:22:38.000000000 +0200
+++ alsaplayer-0.99.80/reader/http/http.c	2009-09-16 01:43:38.000000000 +0200
@@ -731,6 +731,8 @@
     desc->len = 0;
     desc->direction = 0; 
     desc->metadata = NULL;
+    desc->host = NULL;
+    desc->path = NULL;
     desc->status = status;
     desc->data = data;
     pthread_mutex_init (&desc->buffer_lock, NULL);
